prost-build/src/lib.rs

```diff
@@ -949,25 +965,32 @@ impl Config {
         outfile.write_all(format!("{}{}\n", ("    ").to_owned().repeat(depth), line).as_bytes())
     }
 
-    fn generate(&mut self, files: Vec<FileDescriptorProto>) -> Result<HashMap<Module, String>> {
+    /// Processes a set of modules and file descriptors, returning a map of modules to generated
+    /// code contents.
+    ///
+    /// This is generally used when control over the output should not be managed by Prost,
+    /// such as in a flow for a `protoc` code generating plugin. When compiling as part of a
+    /// `build.rs` file, instead use [`compile_protos()`].
+    pub fn generate(
+        &mut self,
+        requests: Vec<(Module, FileDescriptorProto)>,
```

Comment:
module isn't a public type def + seems a bit awkward to expose a `Module` that is just `Vec<String>`. What are your thoughts on making this a struct with limited methods?

Reply:
Yeah I think its fine to keep it in lib.rs

Reply:
Added the `Module` type.